pkg/mcs/scheduling/server/rule/watcher.go

```diff
@@ -216,3 +238,37 @@ func (rw *Watcher) Close() {
 	rw.cancel()
 	rw.wg.Wait()
 }
+
+func (rw *Watcher) hasPendingDeletion() bool {
+	rw.pendingDeletion.RLock()
+	defer rw.pendingDeletion.RUnlock()
+	return len(rw.pendingDeletion.kvs) > 0
+}
+
+func (rw *Watcher) addPendingDeletion(path, groupID, ruleID string) {
+	rw.pendingDeletion.Lock()
+	defer rw.pendingDeletion.Unlock()
+	rw.pendingDeletion.kvs[path] = [2]string{groupID, ruleID}
+}
+
+func (rw *Watcher) tryFinishPendingDeletion() {
```

Comment:
I am worried about if the put and delete can be disordered. If so, the newly added rule might be deleted unexpectedly.